Cython/Build/IpythonMagic.py

```diff
@@ -343,10 +344,21 @@ def critical_function(data):
                 self._profile_pgo_wrapper(extension, lib_dir)
 
         try:
-            self._build_extension(extension, lib_dir, pgo_step_name='use' if args.pgo else None,
+            get_stderr = get_stdout = None
+            with captured_fd(1, sys.getdefaultencoding()) as get_stdout:
+                with captured_fd(2, sys.getdefaultencoding()) as get_stderr:
+                    self._build_extension(extension, lib_dir, pgo_step_name='use' if args.pgo else None,
                                   quiet=args.quiet)
-        except distutils.errors.CompileError:
-            # Build failed and printed error message
+        except (distutils.errors.CompileError, distutils.errors.LinkError):
+            # Build failed, print error message from compiler/linker
```

Comment:
It might be helpful to separate stdout and stderr and give them a header line, so that users know what comes from where.